src/hooks/useProducts.ts

```diff
@@ -0,0 +1,33 @@
+import { keepPreviousData, useQuery } from "@tanstack/react-query";
+import axios from "axios";
+
+const getProducts = async (
+  searchTerm: string,
+  sortBy: string,
+  sortOrder: string,
+  limit: number
+) => {
+  const response = await axios(
+    `https://dummyjson.com/products/search?limit=${limit}&q=${searchTerm}&sortBy=${sortBy}&order=${sortOrder}`
+  );
+
+  if (response.status !== 200) {
+    throw new Error("Unable to fetch data");
+  }
+
+  return response.data.products;
+};
+
+export const useProducts = (
+  searchTerm: string,
+  sortBy: string,
+  sortOrder: string,
+  limit: number
+) => {
+  const query = useQuery({
+    queryKey: ["products", { searchTerm, sortBy, sortOrder, limit }],
+    queryFn: () => getProducts(searchTerm, sortBy, sortOrder, limit),
+    placeholderData: keepPreviousData,
+  });
+  return query;
```

Comment:
suggesiton:
```
  const { data, isError, isFetching } = useQuery({
    queryKey: ["products", { searchTerm, sortBy, sortOrder, limit }],
    queryFn: () => getProducts(searchTerm, sortBy, sortOrder, limit),
    placeholderData: keepPreviousData,
  });

  return { data, isError, isFetching };
};
```
This approach clarifies what’s being returned.
And also good for encapsulation as you can also decide what to return and what to hide from the consumers